app/main/forms/auth_forms.py

```diff
@@ -1,8 +1,20 @@
+from flask import current_app
+from flask_login import current_user
 from flask_wtf import FlaskForm
 from wtforms import PasswordField
-from wtforms.validators import DataRequired, EqualTo, Length, Regexp
+from wtforms.validators import DataRequired, EqualTo, Length, Regexp, ValidationError
+
+from dmutils.email.helpers import hash_string
 from dmutils.forms import StripWhitespaceStringField, StringField
 
+from app import data_api_client
```

Comment:
😿 I've been trying to avoid importing the `data_api_client` outside the apps' `views.py` (so we can keep track of the scope of our globals), but I can't see a better way of doing it for the validators. We do the same thing for  one of the Admin form validators (https://github.com/alphagov/digitalmarketplace-admin-frontend/blob/master/app/main/forms.py#L32). My preference for keeping the validation-related code here in `forms.py` is stronger than my pref for avoiding this import pattern!

Reply:
It is sad indeed to use the globals, but I agree that this is probably better than the alternative :(